flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala

```diff
@@ -922,6 +925,42 @@ class TableSinkITCase extends StreamingTestBase {
 
   }
 
+  @Test
+  def testSinkProvider(): Unit = {
+    val file = tempFolder.newFolder()
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE MyFileSinkTable (
+         |  `a` STRING,
+         |  `b` STRING,
+         |  `c` STRING
+         |) WITH (
+         |  'connector' = 'filesink',
+         |  'path' = '${file.getAbsolutePath}'
+         |)
+         |""".stripMargin)
+
+    val stringTupleData3: Seq[(String, String, String)] = {
+      val data = new mutable.MutableList[(String, String, String)]
+      data.+=(("Test", "Sink", "Hi"))
+      data.+=(("Sink", "Provider", "Hello"))
+      data.+=(("Test", "Provider", "Hello world"))
+      data
+    }
+    val table = env.fromCollection(stringTupleData3).toTable(tEnv, 'a, 'b, 'c)
+    table.executeInsert("MyFileSinkTable").await()
+
+    val source = Source.fromFile(new File(file.getAbsolutePath, file.list()(0)).listFiles()(0).getAbsolutePath)
```

Comment:
If we merge the `filesink` and `filesource` into one connector, we can simply read `MyFileSinkTable` using SELECT query.

Reply:
@wuchong this could only verify the `FileSink` through `Source.fromFile` because the `FileSink` only write a inprogress file which file the `FileSource` doesn't read.

Reply:
@wuchong , this couldn't verify the result by simply reading `MyFileSinkTable` using `SELECT` query. Because the `TestFileTableSink` generates in progress file which file `FileSource` couldn't read.